packages/web3-core-requestmanager/src/jsonrpc.js

```diff
@@ -26,7 +26,9 @@
 
 // Initialize Jsonrpc as a simple object with utility functions.
 var Jsonrpc = {
-    messageId: 0
+    // This is the starting counter for the Jsonrpc.id. And, to insure staying in the safe range while incrementing later,
+    //  pick a random number between 0 and (the maximum safe integer minus the max unsigned 32 bit integer)
+    messageId: Math.floor(Math.random() * (Number.MAX_SAFE_INTEGER - Math.pow( 2, 32 )))
```

Comment:
I had a hunch that this will not fix the actual issue. 

> Responses get mixed up due to conflicting payload IDs

The response will still be mixed up if the payload Ids are conflicting, starting point does not matter and randomizing starting point would minimize but not fixed the problem. 

For once instance of the `web3` there will be no conflicting payload IDs in any case.

Reply:
As per here: https://www.jsonrpc.org/specification

```
An identifier established by the Client that MUST contain a String, Number, or NULL value if included. 
```
`uuid` is already used in 1.x, maybe we could use it, also, for request id`s. We won't need a starting point and other libraries use numbers for ids.

Reply:
Yes, there will be no issue if one instance is running and one library is used. And, yes, this solution will minimize the possibility of a conflict to some grade that is neglected but the possibility is still there.

However, I proposed to handle this in a different way in v4.x as there we can do a breaking change by making the message id as a string that contains a UUID or another format. The new issue is here: https://github.com/ChainSafe/web3.js/issues/5373

And if you have a better suggestion to solve this in v1.x, please advise.

Thanks,

Reply:
We can proceed with this approach but considering it's not a permanent fix. 

Also it does not matter if there is more than one library using the same id for the same endpoint. For the HTTP, the RPC server will be responding synchronously. For WS based RPC server it will be responding per connection. So if there are two libraries, both would be having different connections and that will not conflict the ids.

Reply:
Thanks @nazarhussain ,
As you see this as "not a permanent fix", do you have a better suggestion for 1.x that does not cause a breaking change?

Reply:
> So if there are two libraries, both would be having different connections and that will not conflict the ids.
@nazarhussain I just wanted to note in my original report #5327, I saw a situation where two libraries were using the same connection, which resulted in an ID conflict. I'm not certain how this happened because I'm completely unfamilar with your codebase, but it's somehow possible.

In my case, it was `eth-block-tracker` that was sending a message that somehow bypassed `jsonrpc.js` but still was being processed by the same websocket connection that `jsonrpc.js` was using, which resulted in a conflict.

Maybe someone more experienced than me could figure out how that happened? There might be another lurking bug there.